Sources/PostgresKit/PostgresDataTranslation.swift

```diff
@@ -0,0 +1,365 @@
+                codingPath: self.codingPath + [SomeCodingKey(intValue: self.count)], userInfo: self.encoder.userInfo,
+                value: value, in: self.encoder.context,
+                file: self.encoder.file, line: self.encoder.line
+            ))
+        }
+        mutating func nestedContainer<K: CodingKey>(keyedBy: K.Type) -> KeyedEncodingContainer<K> { self.superEncoder().container(keyedBy: K.self) }
+        mutating func nestedUnkeyedContainer() -> any UnkeyedEncodingContainer { self.superEncoder().unkeyedContainer() }
+        mutating func superEncoder() -> any Encoder { ArrayAwareBoxWrappingPostgresEncoder(
+            codingPath: self.codingPath + [SomeCodingKey(intValue: self.count)], userInfo: self.encoder.userInfo,
+            context: self.encoder.context,
+            file: self.encoder.file, line: self.encoder.line,
+            value: self.encoder.value
+        ) } // NOT the same as self.encoder
+    }
+
+    func encodeNil() throws { self.value.store(scalar: .null) }
+    func encode<T: Encodable>(_ value: T) throws {
+        self.value.store(scalar: try PostgresDataTranslation.encode(
+            codingPath: self.codingPath, userInfo: self.userInfo, value: value, in: self.context, file: self.file, line: self.line
+        ))
+    }
+    
+    struct FallbackSentinel: Error {}
+
+    /// This is a workaround for the inability of encoders to throw errors in various places. It's still better than fatalError()ing.
+    struct FailureEncoder<K: CodingKey>: Encoder, KeyedEncodingContainerProtocol, UnkeyedEncodingContainer, SingleValueEncodingContainer {
+        let codingPath = [any CodingKey](), userInfo = [CodingUserInfoKey: Any](), count = 0
+        init() {}; init() where K == SomeCodingKey {}
+        func encodeNil() throws { throw FallbackSentinel() }
+        func encodeNil(forKey: K) throws { throw FallbackSentinel() }
+        func encode<T: Encodable>(_: T) throws { throw FallbackSentinel() }
+        func encode<T: Encodable>(_: T, forKey: K) throws { throw FallbackSentinel() }
+        func nestedContainer<N: CodingKey>(keyedBy: N.Type) -> KeyedEncodingContainer<N> { .init(FailureEncoder<N>()) }
+        func nestedContainer<N: CodingKey>(keyedBy: N.Type, forKey: K) -> KeyedEncodingContainer<N> { .init(FailureEncoder<N>()) }
+        func nestedUnkeyedContainer() -> any UnkeyedEncodingContainer { self }
+        func nestedUnkeyedContainer(forKey: K) -> any UnkeyedEncodingContainer { self }
+        func superEncoder() -> any Encoder { self }
+        func superEncoder(forKey: K) -> any Encoder { self }
+        func container<N: CodingKey>(keyedBy: N.Type) -> KeyedEncodingContainer<N> { .init(FailureEncoder<N>()) }
+        func unkeyedContainer() -> any UnkeyedEncodingContainer { self }
+        func singleValueContainer() -> any SingleValueEncodingContainer { self }
+    }
+
+    // TODO: Finish all this and make it work
+    /*
```

Comment:
I'm assuming this is all still todo?

Reply:
Yeah, I should yank that stuff, even those bits would have to be redone anyway.